dump/plugins/copp.py

```diff
@@ -0,0 +1,368 @@
+        self.trap_group = ""
+        self.handle_user_and_default_config()
+        self.handle_appl_db()
+        self.handle_asic_db()
+        self.handle_state_db()
+        return self.ret_temp
+    
+    def handle_state_db(self):
+        req = MatchRequest()
+        req.db = "STATE_DB"
+        req.table = "COPP_TRAP_TABLE"
+        req.key_pattern = self.copp_trap_cfg_key
+        ret = self.match_engine.fetch(req)
+        if not ret["error"] and len(ret["keys"]) > 0:
+            self.ret_temp["STATE_DB"]["keys"].append(ret["keys"][0])
+        else:
+            self.ret_temp["STATE_DB"]["tables_not_found"].append("COPP_TRAP_TABLE")
+        
+        req = MatchRequest()
+        req.db = "STATE_DB"
+        req.table = "COPP_GROUP_TABLE"
+        req.key_pattern = self.trap_group
+        ret = self.match_engine.fetch(req)
+        if not ret["error"] and len(ret["keys"]) > 0:
+            self.ret_temp["STATE_DB"]["keys"].append(ret["keys"][0])
+        else:
+            self.ret_temp["STATE_DB"]["tables_not_found"].append("COPP_GROUP_TABLE")
+    
+    def handle_appl_db(self):
+        req = MatchRequest()
+        req.db = "APPL_DB"
+        req.table = APP_COPP_TABLE_NAME
+        req.key_pattern = "*"
+        req.field = "trap_ids"
+        req.value = self.trap_id
+        req.match_entire_list = False
+        req.return_fields = ["trap_group"]
+        ret = self.match_engine.fetch(req)
+        tg = ""
+        if not ret["error"] and len(ret["keys"]) > 0:
+            if len(ret["keys"]) > 1:
+                err_str = "ERROR: Multiple COPP_TABLE Keys found for the trap_id {} in the APPL_DB, keys found: {}".format(self.trap_id, str(ret["keys"]))
+                handle_multiple_keys_matched_error(err_str, ret["keys"][0])
+            self.ret_temp["APPL_DB"]["keys"].append(ret["keys"][0])
+            tg = ret["return_values"][ret["keys"][0]]["trap_group"]
+        else:
+            self.ret_temp["APPL_DB"]["tables_not_found"].append(APP_COPP_TABLE_NAME)
+        
+        if tg != self.trap_group and not self.trap_group and not tg:
+            err_str = "The Associated Trap_group for the trap_id found in APPL and CONFIG_DB/CONFIG_FILE did not match. In APPL_DB: {}, CONFIG_DB: {}".format(tg, self.trap_group)
+            err_str += "\n Proceding with the trap group found in APPL DB"
+            handle_error(err_str, False)
+        
+        if tg:
+            self.trap_group = tg
+
+        
+    def handle_asic_db(self):
+        if self.trap_id not in trap_id_map:
+            handle_error("Trap Id {} is not present in the map".format(self.trap_id), False)
+            sai_trap_id = ""
+        else:
+            sai_trap_id = trap_id_map[self.trap_id]
+        sai_trap_grp = self.__get_asic_trap_obj(sai_trap_id)
+        sai_queue, sai_policer = self.__get_asic_trap_group_obj(sai_trap_grp)
+        self.__get_asic_policer_obj(sai_policer)
+        self.__get_asic_queue_obj(sai_queue)
+        
```

Comment:
The currently implementation does not customize the search based on the trap_id passed. If this is something which can be done for all the trap_id's, then this can be implemented. 

As this is specific to sflow, i think this should be in the implementation of sflow dump module rather than in COPP.